db/schema_tables.cc

```diff
@@ -2222,11 +2222,53 @@ std::vector<mutation> make_drop_aggregate_mutations(schema_features features, sh
  * Table metadata serialization/deserialization.
  */
 
+/// Returns mutations which when applied to the database will cause all schema changes
+/// which create or alter the table with a given name, made with timestamps smaller than t,
+/// have no effect. Used when overriding schema to shadow concurrent conflicting schema changes.
+static schema_mutations make_table_deleting_mutations(const sstring& ks, const sstring& table, bool is_view, api::timestamp_type t) {
+    tombstone tomb;
+
+    // Generate neutral mutations if t == api::min_timestamp
+    if (t > api::min_timestamp) {
+        tomb = tombstone(t - 1, gc_clock::now());
+    }
+
+    auto tables_m_s = is_view ? views() : tables();
+    mutation tables_m{tables_m_s, partition_key::from_singular(*tables_m_s, ks)};
+    {
+        auto ckey = clustering_key::from_singular(*tables_m_s, table);
+        tables_m.partition().apply_delete(*tables_m_s, ckey, tomb);
+    }
+
+    mutation scylla_tables_m{scylla_tables(), partition_key::from_singular(*scylla_tables(), ks)};
+    {
+        auto ckey = clustering_key::from_singular(*scylla_tables(), table);
+        scylla_tables_m.partition().apply_delete(*scylla_tables(), ckey, tomb);
+    }
+
+    auto make_drop_columns = [&] (const schema_ptr& s) {
+        mutation m{s, partition_key::from_singular(*s, ks)};
+        auto ckey = clustering_key::from_exploded(*s, {utf8_type->decompose(table)});
```

Comment:
The schema object is different, and key objects are schema-dependent so we shouldn't reuse them for different tables. It may currently work if the schema is logically the same, but is a violation which may bite later.

Reply:
Oh, maybe you're referring to the fact that we're using from_exploded. This is needed because we're making a prefix key here, it has two columns unlike in other tables. The column name is the last component which we're not specifying so that all columns are covered by the Tombstone.